perf/perf_24x7_all_events.py

```diff
@@ -0,0 +1,135 @@
+#!/usr/bin/env python
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+#
+# Copyright: 2019 IBM
+# Author: Nageswara R Sastry <rnsastry@linux.vnet.ibm.com>
+
+import os
+import platform
+from avocado import Test
+from avocado import main
+from avocado.utils import cpu, distro, memory, process
+from avocado.utils.software_manager import SoftwareManager
+
+
+class hv_24x7_all_events(Test):
+
+    """
+    This tests all hv_24x7 events
+    """
+    # Initializing fail command list
+    fail_cmd = list()
+
+    def setUp(self):
+        """
+        Setup checks :
+        0. Processor should be ppc64.
+        1. Perf package
+        2. 24x7 is not supported on guest
+        3. 24x7 is present
+        4. Performance measurement is enabled in LPAR through BMC
+        """
+        smm = SoftwareManager()
+        detected_distro = distro.detect()
+        if 'ppc' not in process.system_output("uname -p", ignore_status=True):
```

Comment:
distro.arch can be used and even string also need be changed as ppc64 stands for BE version 

like 
```
if 'ppc' not in detected_distro.arch:
    self.cancel("Processor is not PowerPC")
```

Reply:
done